### ISSUE TITLE
tool.vcvars_dict failed if filter_known_paths=True

### ISSUE THREAD

**Issue by dvetutnev** [conan-io/conan#3940]
[Failed this](https://github.com/conan-io/conan/blob/1.9.1/conans/client/tools/win.py#L460)

OS Windows 7, Conan version 1.9.1, Python 2.7.15
(also class 'list' don`t have method 'split' on Python 3)

My code (in conanfile.py):

    def get_build_environment(self):
        env = {}
        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
            try:
                env = tools.vcvars_dict(self.settings, filter_known_paths=True, force=True)
            except:
                traceback.print_exc()
                raise
            toolset = str(self.settings.compiler.get_safe("toolset"))
            if toolset.endswith("_xp"):
                import find_sdk_winxp
                env = find_sdk_winxp.dict_append(self.settings.arch, env=env)
        return env

Console log:
```
boost/1.68.0@odant/tt: Calling build()
boost/1.68.0@odant/tt: -------------- Bootstrap ------------------------
Traceback (most recent call last):
  File "D:\.conan\data\boost\1.68.0\odant\tt\export\conanfile.py", line 171, in get_build_environment
    env = tools.vcvars_dict(self.settings, filter_known_paths=True, force=True)
  File "D:\Python27\lib\site-packages\conans\client\tools\win.py", line 460, in vcvars_dict
    path = new_env.get("PATH", "").split(";")
AttributeError: 'list' object has no attribute 'split'
boost/1.68.0@odant/tt:
boost/1.68.0@odant/tt: ERROR: Package '74d6e6f7d63123455b1033138a5a462ab0736c21' build failed
boost/1.68.0@odant/tt: WARN: Build folder D:/.conan/s\4w_jf4\1
ERROR: boost/1.68.0@odant/tt: Error in build() method, line 80
        b2 = self.bootstrap(source_folder)
while calling 'bootstrap', line 101
        env = self.get_build_environment()
while calling 'get_build_environment', line 171
        env = tools.vcvars_dict(self.settings, filter_known_paths=True, force=True)
        AttributeError: 'list' object has no attribute 'split'
```

**Comment by lasote** [comment]
Thanks, I'm on it, we will try to release the `1.9.2` with this bug fixed next week probably.

**Comment by danimtb** [comment]
This was closed in #3941 and available in Conan 1.9.2
